rpcserver.go

```diff
@@ -5067,3 +5075,108 @@ func (r *rpcServer) SubscribeChannelBackups(req *lnrpc.ChannelBackupSubscription
 		}
 	}
 }
+
+// ChannelAcceptor dispatches a bi-directional streaming RPC in which
+// OpenChannel requests are sent to the client and the client responds with
+// a boolean that tells LND whether or not to accept the channel. This allows
+// node operators to specify their own criteria for accepting inbound channels
+// through a single persistent connection. There may only be one client at a
+// time for this RPC.
+func (r *rpcServer) ChannelAcceptor(stream lnrpc.Lightning_ChannelAcceptorServer) error {
+
+	var (
+		acceptorClient *chanacceptor.Client
+		rpcAcceptor    *chanacceptor.RPCAcceptor
+		ok             bool
+	)
+
+	// Locate the RPCAcceptor within the ChainedAcceptor and register this RPC client.
+	openChanPredicate := r.server.fundingMgr.cfg.OpenChannelPredicate
+	chainedAcceptor, _ := openChanPredicate.(*chanacceptor.ChainedAcceptor)
+
+	for _, acceptor := range chainedAcceptor.Acceptors {
+		rpcAcceptor, ok = acceptor.(*chanacceptor.RPCAcceptor)
+		if ok {
+			tempClient, err := rpcAcceptor.RegisterClient()
+			if err != nil {
+				return err
+			}
+
+			acceptorClient = tempClient
+			break
+		}
+	}
+
+	if acceptorClient == nil {
+		return errors.New("unable to set up stream")
+	}
+
+	// errChan is used by the receive loop to signal any errors that occur
+	// during reading from the stream. This is primarily used to shutdown the
+	// send loop in the case of an RPC client disconnecting.
+	errChan := make(chan error)
+
+	// We need to have the stream.Recv() in a goroutine since the call is
+	// blocking and would prevent us from sending more OpenChannelRequests to
+	// the RPC client.
+	go func() {
+		defer rpcAcceptor.UnregisterClient()
+		for {
+			resp, err := stream.Recv()
+			if err != nil {
+				select {
+				case errChan <- err:
+				case <-r.quit:
+				}
+				return
+			}
+
+			var pendingID [32]byte
+			copy(pendingID[:], resp.PendingChanId)
+
+			openChanResp := chanacceptor.OpenChannelResponse{
+				Accept:           resp.Accept,
+				PendingChannelID: pendingID,
+			}
+
+			// Now that we have the response from the RPC client, send it to
+			// the acceptorClient's RecvChan.
+			select {
+			case acceptorClient.RecvChan <- openChanResp:
+			case <-r.quit:
+				return
+			}
+		}
+	}()
+
+	for {
+		select {
+		case req := <-acceptorClient.SendChan:
```

Comment:
These channel attributes (to be morphed into methods based on the comments above), would be better named as response/request rather than send/recv. IMO the intent is more clear with the request/response naming.